src/lib/duration/create.js

```diff
@@ -45,14 +45,21 @@ export function createDuration (input, key) {
         };
     } else if (!!(match = isoRegex.exec(input))) {
         sign = (match[1] === '-') ? -1 : 1;
+        var ySign = (match[2] === '-') ? -sign : sign;
```

Comment:
I think it can be simplified by making the parseFloat handle the sign. If the isoRegex includes `(?:-)?` inside each of the elements before the character class, then it can be part of the capture group and will be parsed as negative by parseFloat. I can update the pull request to include that.